src/shared/sleep-config.c

```diff
@@ -266,7 +281,11 @@ int can_sleep(const char *verb) {
 
         assert(streq(verb, "suspend") ||
                streq(verb, "hibernate") ||
-               streq(verb, "hybrid-sleep"));
+               streq(verb, "hybrid-sleep") ||
+               streq(verb, "suspend-to-hibernate"));
+
+        if (streq(verb, "suspend-to-hibernate"))
+                return can_sleep("suspend") && can_sleep("hibernate");
```

Comment:
Shouldn't this check for /sys/class/rtc/rtc0/wakealarm availability?

Reply:
Yes, very good point.  I'll add that.